### PR TITLE
BUG: Fix melt with mixed int/str columns

### LINKED ISSUE #29718: melt does not recognize numeric column names
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

df = pd.DataFrame(columns=[1, "string"])
pd.melt(df, id_vars=[1, "string"])
```
#### Problem description

The shown example fails with 
```
Traceback (most recent call last):
  File "test.py", line 5, in <module>
    pd.melt(df, id_vars=[1, "string"])
  File "/home/nils/projects/tsfresh/venv/lib/python3.6/site-packages/pandas/core/reshape/melt.py", line 52, in melt
    "".format(missing=list(missing))
KeyError: "The following 'id_vars' are not present in the DataFrame: ['1']"
```
and I guess the reason is that the call of
```python
Index(np.ravel(id_vars))
```
in `pd.melt` somehow casts the numerical column name `1` to the string `"1"`. 

I am not sure if this is intended behavior or if the case of numerical column names is just not supported, but at least in older pandas versions (e.g. 0.23.4) this still worked.

Thanks for looking into this! I am also fine if this is closed with "won't fix" :-)

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.6.8.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-65-generic
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 0.25.3
numpy            : 1.17.4
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 19.3.1
setuptools       : 41.6.0
Cython           : None
pytest           : 5.2.4
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.4.1
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.10.3
IPython          : 7.9.0
pandas_datareader: 0.8.1
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.4.1
matplotlib       : 3.1.1
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : 1.3.2
sqlalchemy       : None
tables           : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None

</details>

### PR DESCRIPTION
**Pull request by jschendel**
- [X] closes #29718
- [X] tests added / passed
- [X] passes `black pandas`
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

### CODE REVIEW

File: pandas/tests/reshape/test_melt.py [hunk 1]
```diff
@@ -289,34 +289,53 @@ def test_melt_missing_columns_raises(self):
         df = pd.DataFrame(np.random.randn(5, 4), columns=list("abcd"))
 
         # Try to melt with missing `value_vars` column name
-        msg = "The following '{Var}' are not present in the DataFrame: {Col}"
-        with pytest.raises(
-            KeyError, match=msg.format(Var="value_vars", Col="\\['C'\\]")
-        ):
+        msg = r"The following '{var}' are not present in the DataFrame: \[{col}\]"
```
Comment (jschendel) [inline pandas/tests/reshape/test_melt.py]: Made a couple small cleanups to the error message not directly related to the bug in question:
- Added brackets around `{col}` so they didn't need to be manually added to each message, reducing a few multi-line statements to a single line
- Made the variables lowercase to be more stylistically consistent with other usages of `format`

**Comment by jbrockmendel** [comment]
LGTM

**Comment by nils-braun** [comment]
Thanks for fixing so quickly!

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Great thanks @jschendel

**Pull request merged**